### PR TITLE
[runtime env] plugin refactor[6/n]: java api refactor

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?

The related prototype https://github.com/ray-project/ray/pull/25387

### CODE REVIEW

File: java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java [hunk 1]
```diff
@@ -1,38 +1,30 @@
...
+import io.ray.api.exception.RuntimeEnvException;
 
 /** This is an experimental API to let you set runtime environments for your actors. */
 public interface RuntimeEnv {
 
-  String toJsonBytes();
+  void set(String name, Object value);
 
-  public static class Builder {
+  public void setJsonStr(String name, String jsonStr) throws RuntimeEnvException;
```
Comment (jovany-wang) [inline java/api/src/main/java/io/ray/api/runtimeenv/RuntimeEnv.java:11]: It's better to comment for `setJsonStr` and `getJsonStr` due to they are not straightforward.

File: java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java [hunk 2]
```diff
@@ -72,6 +75,8 @@ public abstract class AbstractRayRuntime implements RayRuntime {
 
   private static ParallelActorContextImpl parallelActorContextImpl = new ParallelActorContextImpl();
 
+  private static ObjectMapper mapper = new ObjectMapper();
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java:78]:
```suggestion
  private final static ObjectMapper MAPPER = new ObjectMapper();
```

File: java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java [hunk 3]
```diff
@@ -1,63 +1,93 @@
...
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.io.IOException;
 
 public class RuntimeEnvImpl implements RuntimeEnv {
 
-  private Map<String, String> envVars = new HashMap<>();
+  private static ObjectMapper mapper = new ObjectMapper();
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java:17]:
```suggestion
  private final static ObjectMapper MAPPER = new ObjectMapper();
```

File: java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java [hunk 4]
```diff
@@ -1,63 +1,93 @@
...
+    }
     JsonFormat.Printer printer = JsonFormat.printer();
     try {
       return printer.print(protoRuntimeEnvInfoBuilder);
     } catch (InvalidProtocolBufferException e) {
       throw new RuntimeException(e);
     }
   }
+
+  public RuntimeEnvCommon.RuntimeEnvInfo GenerateRuntimeEnvInfo() throws RuntimeEnvException {
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java:81]: There is no any usage for `GenerateRuntimeEnvInfo`.

Comment (SongGuyang) [reply java/runtime/src/main/java/io/ray/runtime/runtimeenv/RuntimeEnvImpl.java:81]: Here https://github.com/ray-project/ray/pull/26783/files#diff-de3a842e353216ddd6d48f5eb9da067f4c3936a7444f1df64945f6a4cc1baf25R111 ?

**Review by jovany-wang — APPROVED** [review]

**Comment by SongGuyang** [comment]
The failed tests is not related.

**Pull request merged**